libraries/support/azure/deprecations/strings_in_where_clause.rb

```diff
@@ -0,0 +1,26 @@
+module Deprecations
+  module StringsInWhereClause
+    def self.included(base)
+      base.class_eval do
+        alias_method :old_where, :where
+
+        def where(conditions = {}, &block)
+          transformed = {}
+          conditions.each do |key, value|
+            if key.is_a? String
+              warn '[DEPRECATION] String detected in where clause. As of verison 1.2 ' \
+                'where clauses should use a symbol. Please convert ' \
+                "where('#{key}' => '#{value}') to where(#{key}: '#{value}'). " \
+                'Automatic conversion will be removed in version 2.0.'
+
+              transformed[key.to_sym] = value
+            else
+              transformed[key] = value
+            end
+          end
+          old_where(transformed, &block)
+        end
```

Comment:
We can break this into simpler statements for easier comprehension.

Here we emit the deprecation if needed, but always ensure the key is a symbol, whether it was a string or already a symbol.

Also note spelling fix `s/verison/version/` in warning.
```ruby
def where(conditions = {}, &block)
  string_warn = ->(k, v) { warn_deprecation(k, v) if k.is_a?(String); [:"#{k}", v] }

  old_where(conditions.map(&string_warn).to_h, &block)
end

private

def warn_deprecation(key, value)
  warn '[DEPRECATION] String detected in where clause. As of version 1.2 ' \
    'where clauses should use a symbol. Please convert ' \
    "where('#{key}' => '#{value}') to where(#{key}: '#{value}'). " \
    'Automatic conversion will be removed in version 2.0.'
end
```

Reply:
👍 That's much easier to read. Thanks!